src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/FileIO/FileSystem.Portable.vb

```diff
@@ -0,0 +1,1522 @@
+            VerifyUICancelOption("onUserCancel", onUserCancel)
+
+            ' Get the full path. This will handle invalid paths exceptions.
+            Dim directoryFullPath As String = IO.Path.GetFullPath(directory)
+
+            ' Demand Write permission for security reason (see CopyOrMoveFile / CopyOrMoveDirectory).
+            'DemandDirectoryPermission(directoryFullPath, FileIOPermissionAccess.Write)
+
+            ' Throw if device path.
+            ThrowIfDevicePath(directoryFullPath)
+
+            If Not IO.Directory.Exists(directoryFullPath) Then
+                Throw ExUtils.GetDirectoryNotFoundException(SR.IO_DirectoryNotFound_Path, directory)
+            End If
+
+            ' VsWhidbey 319809: Throw exception if deleting root directory.
+            If IsRoot(directoryFullPath) Then
+                Throw ExUtils.GetIOException(SR.IO_DirectoryIsRoot_Path, directory)
+            End If
+
+            ' If user want shell features (Progress, Recycle Bin), call shell operation.
+            ' We don't need to consider onDirectoryNotEmpty here (VSWhidbey 283409).
+            If (showUI <> UIOptionInternal.NoUI) AndAlso Environment.UserInteractive Then
+#If HaveUI Then
+                ShellDelete(directoryFullPath, showUI, recycle, onUserCancel, FileOrDirectory.Directory)
+#Else
+                Throw New PlatformNotSupportedException("No UI for ShellDelete(directoryFullPath, showUI, recycle, onUserCancel, FileOrDirectory.Directory)")
+#End If
+                Exit Sub
+            End If
+
+            ' Otherwise, call Framework's method.
+            IO.Directory.Delete(directoryFullPath, onDirectoryNotEmpty = DeleteDirectoryOption.DeleteAllContents)
+        End Sub
+
+        Private Shared Sub DeleteFileInternal(ByVal file As String, ByVal showUI As UIOptionInternal, ByVal recycle As RecycleOption,
+                    ByVal onUserCancel As UICancelOption)
+
+            ' Verify enums
+            VerifyRecycleOption("recycle", recycle)
+            VerifyUICancelOption("onUserCancel", onUserCancel)
+
+            ' Get the full path. This will handle invalid path exceptions.
+            Dim fileFullPath As String = NormalizeFilePath(file, "file")
+
+            ' Demand Write permission for security reason (see CopyOrMoveFile / CopyOrMoveDirectory).
+            'Call (New FileIOPermission(FileIOPermissionAccess.Write, fileFullPath)).Demand()
+
+            ' Throw if device path.
+            ThrowIfDevicePath(fileFullPath)
+
+            If Not IO.File.Exists(fileFullPath) Then
+                Throw ExUtils.GetFileNotFoundException(file, SR.IO_FileNotFound_Path, file)
+            End If
+
+            ' If user want shell features (Progress, Recycle Bin), call shell operation.
+            If (showUI <> UIOptionInternal.NoUI) AndAlso Environment.UserInteractive Then
+#If HaveUI Then
+                ShellDelete(fileFullPath, showUI, recycle, onUserCancel, FileOrDirectory.File)
```

Comment:
All these ShellXXX functions should be OK to wire up. They call the Windows shell, not Winforms. Yes, they may pop OS UI but that's OK.

Unix will have to have its own path -- you'll want to have partial classes across FileSystem.vb, FileSystem.Windows.vb, and FileSystem.Unix.vb. ShellXX API are Windows only, so any methods that call those will need counterparts in the Unix file that throw PlatformNotSupportedException.

Reply:
Of course, we cannot normally unit test any API that pops UI. We do need tests though! What we do for cases that need manual intervention is we decorate them with `[ConditionalTheory(nameof(ManualTestsEnabled))]` and `ManualTestsEnabled` is set elsewhere by an environment variable -- look in the Console tests for example.

Reply:
I will look into this this week, I want to get all the test to pass first and add special cases for where VB is different than IO.*, this comment applies to next comment as well. Not sure how to handle smaller limits in VB that don't match IO right now I am just adding checks that they fail.

Reply:
After looking there are large pieces of code deleted from the original source that we require significant time to reconstruct to get Windows UI working. I would like to put that in a separate PR. I will put the tests for it around [ConditionalTheory(nameof(ManualTestsEnabled))]   for now and they can be ignored.

Reply:
I have created two new fines FileSystem.Windows.VB and FileSystem.Unix.VB with the different OS specific UI Code. How do I get just the Windows or Unix version to Build so I don't get multiple definitions?